### ISSUE TITLE
Completely Missing Nearly Half of My SQL Notebook After a Crash

### ISSUE THREAD

**Issue by rayswebpresence** [microsoft/azuredatastudio#15846]
Issue Type: <b>Bug</b>

I have no idea what happened; I can tell that it wasn't an issue with saving because cells I have edited as recently as today exist, while cells lower in the notebook which had been created last week disappeared.

Unfortunately lost a great deal of work. All I recall from the error message was "OOM" - it also suggested I shouldn't worry as my work would reopen right away.

You can imagine my surprise when it reopened and most of that work was gone.

Azure Data Studio version: azuredatastudio 1.30.0 (59c4b8e90cf2d3a24bed72623197f10f8d090fdc, 2021-06-16T19:04:08.858Z)
OS version: Windows_NT x64 10.0.19043

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz (12 x 3696)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.92GB (15.85GB free)|
|Process Argv||
|Screen Reader|yes|
|VM|0%|
</details><details><summary>Extensions (25)</summary>

Extension|Author (truncated)|Version
---|---|---
better-comments|aar|2.1.0
firstresponderkit|dre|0.5.0
newdatabase|kev|0.0.3
git-graph|mhu|1.28.0
admin-pack|Mic|0.0.2
admin-tool-ext-win|Mic|0.0.2
agent|Mic|0.48.0
azuredatastudio-postgresql|Mic|0.2.7
cms|Mic|0.8.0
datavirtualization|Mic|1.12.0
import|Mic|1.3.0
kusto|Mic|0.5.4
machine-learning|Mic|0.10.0
powershell|ms-|2020.4.0
profiler|Mic|0.11.0
query-history|Mic|0.1.0
schema-compare|Mic|1.11.0
server-report|Mic|0.2.3
sql-assessment|Mic|0.6.1
sql-database-projects|Mic|0.10.0
whoisactive|Mic|0.1.4
azdata-sanddance|msr|3.2.0
sql-search|Red|0.3.1
synthwave-vscode|Rob|0.1.8
vscodeintellicode|Vis|1.2.1

(3 theme extensions excluded)

</details>

**Comment by VasuBhog** [comment]
Hi @rayswebpresence, we are sorry to hear that. In order to understand the issue, we would like to know more about the notebook. 
How large was the notebook file?
Are you able to share it with us?

**Comment by rayswebpresence** [comment]
It wasn't that large - maybe a dozen cells all together.

Each one returned (roughly) 30-300k thousand cells - though they limited it to 5k in the notebook probably to minimize issues.

The notebook as it is now (only 2 cells plus 4 I have since recreated) only showing up as 2,276,812 bytes.

I'm unable to share the file because it is work related. 

If you're unable to move forward without the notebook, that's okay. 

In case it's one of my installed extensions, I've moved it to source control. and I will be in touch if it reoccurs.

Assuming it happens again, I'll try to anonymize it enough so I can share.
Let me know if there's any steps I can other than sharing the notebook itself.
